src/main/java/com/sarality/db/AddIndexSchemaUpdate.java

```diff
@@ -0,0 +1,54 @@
+package com.sarality.db;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Schema Update operation for adding a new Index to an existing table.
+ *
+ * @author abhideep@ (Abhideep Singh)
+ */
+public class AddIndexSchemaUpdate implements SchemaUpdate {
+
+  private final String tableName;
+  private final String indexName;
+  private final List<Column> columns;
+  private final Integer dbVersion;
+
+  //primary and required columns not supported currently
+  public AddIndexSchemaUpdate(Integer dbVersion, String tableName, String indexName, Column column, Column... columns) {
```

Comment:
I'm assuming we don't need Unique indexes or Sort order support right now and we'll worry about it later.

Reply:
Not for now. When we need it, we'll add a TableIndex just like TableColumn